mozetl/clientsdaily/rollup.py

```diff
@@ -0,0 +1,121 @@
+import datetime as DT
+from pyspark.sql import SparkSession
+from pyspark.sql import functions as F
+import click
+from moztelemetry.standards import filter_date_range
+from mozetl.utils import generate_filter_parameters, format_spark_path
+from fields import MAIN_SUMMARY_FIELD_AGGREGATORS
+
+ACTIVITY_SUBMISSION_LAG = DT.timedelta(10)
+MAIN_SUMMARY_VERSION = 4
+MAIN_SUMMARY_PATH = "s3://telemetry-parquet/main_summary/v{}".format(
+    MAIN_SUMMARY_VERSION)
+WRITE_VERSION = '2'
+STORAGE_BUCKET = 'net-mozaws-prod-us-west-2-pipeline-analysis'
+STORAGE_SUB_DIR = '/spenrose/clients-daily/v{}/'.format(WRITE_VERSION)
+
+
+def load_main_summary(spark):
+    return spark.read.option(
+        "mergeSchema", "true").parquet(MAIN_SUMMARY_PATH)
+
+
+def extract_search_counts(frame):
+    '''
+    The result should have exactly as many rows as the input.
+    '''
+    two_columns = frame.select(F.col("document_id").alias("did"), "search_counts")
+    exploded = two_columns.select(
+        "did", F.explode("search_counts").alias("search_struct"))
+    unpacked = exploded.select(
+        "did",
+        F.expr("search_struct.count").alias("search_count_atom")
+    )
+    grouped = unpacked.groupBy("did").agg({"search_count_atom": "sum"})
+    extracted = grouped.select(
+        "did", F.col("sum(search_count_atom)").alias("search_count"))
+    nulls = two_columns.select(
```

Comment:
Please reformat this again, either nested in a new context or with the closing parenthesis on a new line.